awesomplete.js

```diff
@@ -262,11 +270,26 @@ _.ITEM = function (text, input) {
 };
 
 _.REPLACE = function (text) {
-	this.input.value = text;
+	this.input.value = text.value;
 };
 
+_.DATA = function (item/*, input*/) { return item; };
+
 // Private functions
 
+function Suggestion(data) {
+	var o = Array.isArray(data)
+	  ? { label: data[0], value: data[1] }
+	  : typeof data === "object" ? data : { label: data, value: data };
+
+	this.label = o.label;
+	this.value = o.value;
+}
+Suggestion.prototype = new String;
```

Comment:
Changed to `Object.create(String.prototype)` and provided correct `length` in 61666cf
Sure, length is not live updated, but then we can also modify `value` and `label` while pretending to be just a String.

So should we care about not being exactly the same as String here?
I believe it's not fully doable in ES5.

Will we eventually remove Suggestion in 2.0 or do you want to keep it forever?

I'll check the link you provided and explore other solutions as well.